src/status_im/ui/components/connectivity/view.cljs

```diff
@@ -133,29 +157,84 @@
+                     (not
+                      (or (:connected? status-properties)
+                          (:connecting? status-properties))))
+                    false
+
+                    :else
+                    true)]
+    (if enqueue?
+      (when (or (and (nil? @enqueued-connectivity-status-properties)
+                     (not= status-properties current-ui-status-properties))
+                (and (some? @enqueued-connectivity-status-properties)
+                     (not= status-properties @enqueued-connectivity-status-properties)))
+        ;; reset queued with new state and start a timer if not yet started
+        (reset! enqueued-connectivity-status-properties status-properties)
+        (when-not @timer
+          (reset! timer (utils/set-timeout #(do
+                                              (reset! timer nil)
+                                              (when @enqueued-connectivity-status-properties
+                                                (re-frame/dispatch [:set :connectivity/ui-status-properties @enqueued-connectivity-status-properties])
+                                                (reset! enqueued-connectivity-status-properties nil)))
+
+                                           ;; if the app is in foreground for less than 5s, postpone state changes for 5s otherwise 1s
+                                           (if
+                                            (and app-active-since
+                                                 (< (- (datetime/timestamp) app-active-since)
+                                                    5000))
+                                             5000
+                                             1000)))))
+      (when (not= status-properties current-ui-status-properties)
+        ;; send immediately
+        (reset! enqueued-connectivity-status-properties nil)
+        (re-frame/dispatch [:set :connectivity/ui-status-properties status-properties])
+        (when @timer
+          (utils/clear-timeout @timer)
+          (reset! timer nil))))))
 
 (defview connectivity-view [anim-translate-y]
   (letsubs [status-properties [:connectivity/status-properties]
+            app-active-since  [:app-active-since]
+            current-ui-status-properties  [:connectivity/ui-status-properties]
+            status-hidden (reagent/atom true)
             view-id           [:view-id]
             window-width (reagent/atom 0)]
-    {:component-did-mount
-     (fn []
-       (when anim-translate-y
-         (if (:connected? status-properties)
-           (animation/set-value anim-translate-y neg-connectivity-bar-height)
-           (animation/set-value anim-translate-y 0))))}
-    (let [{:keys [loading-indicator?]} status-properties]
+    (let [loading-indicator? (:loading-indicator? current-ui-status-properties)
+          _ (propagate-status status-properties app-active-since current-ui-status-properties)]
```

Comment:
you must use react  lifecycle methods like `component-did-update`

Reply:
@flexsurfer doesn't work very well: i need to run that code every time one of the letsub changes.
lifecycle fires only if view's props change, right?

Reply:
in theory it should work , but if not you could create new component and pass parameters